woocommerce/woocommerce-ios
# Convert async -> Result signature functions to async throws

Closes: #8104 (Sorry for the bigger PR)


### Description

With this PR I convert the function signatures with `await -> Result<..., Error>` to the most natural `async throws -> ...`, except in those cases where we have a specific error type e.g [here](https://github.com/woocommerce/woocommerce-ios/blob/636a27307afac5c2cc1ac7cb45dc349172c08c23/WooCommerce/Classes/Authentication/Store%20Creation/StoreCreationCoordinator.swift#L220)

### Testing instructions

Unit tests should pass. You can also check that the related features do not present any side effects:
- Remove Apple Id access
- Create account
- Domain suggestions
- Username suggestions
- Free domain suggestions
- Site creation
- Just in Time Messages
- Stats


### Screenshots



---
